### PR TITLE
Updates to some TF notebooks because data collators are now mandatory

### PR DESCRIPTION
**Pull request by Rocketknight1**

### CODE REVIEW

**Comment by Rocketknight1** [comment]
Also, I know reviewing notebooks on Github is very annoying - the only change here is that data collators are now being used in all examples, whereas before some relied on the default behaviour of to_tf_dataset() when one wasn't provided, which in retrospect should never have been there.

File: examples/language_modeling_from_scratch-tf.ipynb [hunk 1]
```diff
@@ -1198,26 +1218,28 @@
      "name": "stderr",
      "output_type": "stream",
      "text": [
-      "Cloning https://huggingface.co/Rocketknight1/bert-base-cased-finetuned-wikitext2 into local empty directory.\n"
+      "/home/matt/miniconda3/envs/tensorflow26/lib/python3.9/site-packages/huggingface_hub/hf_api.py:715: FutureWarning: `create_repo` now takes `token` as an optional positional argument. Be sure to adapt your code!\n",
+      "  warnings.warn(\n",
+      "/home/matt/PycharmProjects/notebooks/examples/mlm_from_scratch_model_save is already a clone of https://huggingface.co/Rocketknight1/bert-base-cased-finetuned-wikitext2. Make sure you pull the latest changes with `repo.git_pull()`.\n"
```
Comment (LysandreJik) [inline examples/language_modeling_from_scratch-tf.ipynb:1223]: Maybe some warnings like this one can be removed for clarity/readability

**Review by LysandreJik — APPROVED** [review]
Looks good, thank you @Rocketknight1

File: examples/language_modeling_from_scratch-tf.ipynb (hunk 1 above)
Comment (Rocketknight1) [reply examples/language_modeling_from_scratch-tf.ipynb:1223]: Those warnings started being fired recently by the `PushToHubCallback`, although it still works correctly. Do you want me to leave merging this until I investigate that, or should I just merge this now and put that on the list to be fixed when I can? It's only cosmetic, as far as I can tell.

Comment (LysandreJik) [reply examples/language_modeling_from_scratch-tf.ipynb:1223]: You can merge it now!

Comment (Rocketknight1) [reply examples/language_modeling_from_scratch-tf.ipynb:1223]: Sure!

**Pull request merged**